float64 curTime
float64 curLen
float64 curDesLen
